lib/BaseClient.ts

```diff
@@ -0,0 +1,25 @@
+import Logger from './Logger';
+
+abstract class BaseClient {
+  logger: Logger;
+  status!: ClientStatus;
+
+  constructor() {
+    this.logger = Logger.global;
+  }
+
+  setStatus(val: ClientStatus): void {
+    this.logger.info(`${this.constructor.name} status: ${val}`);
+    this.status = val;
+  }
+
+  isDisabled(): boolean {
+    return this.status === ClientStatus.DISABLED;
+  }
+}
+
+export default BaseClient;
+export enum ClientStatus {
```

Comment:
Let's stick to defining enums/interfaces before the class, for an easy read. 
If it's also for export, we'll do it in the end of the file, as usual (`export {ClientStatus}`)

Reply:
Also, let's stick with singular enum types from now (as you already did).